### DISCUSSION TITLE
Cookie kiali-token-openid-nonce rjected, as expired

### DISCUSSION THREAD

**Discussion by schroedt** [kiali/kiali#3571] [Q&A] [answered]
Hi Again,

After integrating Keycloak with `Jenkins` and `Gitbucket` and some re-configurations inside Jenkins, I noticed an error while logging into kiali. Browser console says: `Cookie kiali-token-openid-nonce rjected, as expired` and I receive a HTTP 401 Unauthorized. As I needed to re-deploy Keycloak a few times (data is still available and login works with all other workloads, token is generated at keycloak), I assume used to correlated to K8s or Kiali.

![grafik](https://user-images.githubusercontent.com/6430367/104103320-b849f900-52a1-11eb-9fc1-13979b7d989c.png)

No information within any logs (neither `Keycloak` nor `istio-sytem`/`kiali`).
Any idea how to debug this issue?

**Comment by schroedt** [comment, marked as answer]
Hi there,

I found the issue. My reverse-proxy (Nginx) seem to have interfered the connectivity. Skipping Nginx made everthing fine.

Does anyone know which configurations at reverse-proxies are recommended for forwarding token information for Kiali login?

Further: If you don't trust cookies, logs throw certain exceptions. I recommend trusting your Kiali installation for proper functionality.

**Reply by jmazzitelli** [reply]
> Does anyone know which configurations at reverse-proxies are recommended for forwarding token information for Kiali login?

See [this issue](https://github.com/kiali/kiali/issues/3207) and the PRs that are currently being reviewed. If you can, test that in your reverse proxy setup to see if those PRs work for you (would be good to get confirmation from someone else those PRs successfully support reverse proxies) - I believe the author of the PRs has published some test images to docker hub for others to test with.

**Reply by schroedt** [reply]
Hi @jmazzitelli,

thanks for pinging. I will check and update #3207.
